ocaml/xapi/xapi_pool.ml

```diff
@@ -143,6 +143,41 @@ let pre_join_checks ~__context ~rpc ~session_id ~force =
     end
   in
 
+  let assert_homogeneous_updates () =
+    let module S  = Helpers.StringSet in
+    let localhost = Helpers.get_localhost ~__context in
+    let updates_of host = (* patches on pool host *)
+      Client.Host.get_updates ~rpc ~session_id ~self:host
+      |> List.map (fun self -> Client.Pool_update.get_record ~rpc ~session_id ~self)
+      (* filter down to patches that are relevant *)
+      |> List.filter (fun upd -> upd.API.pool_update_enforce_homogeneity = true)
+      |> List.map (fun upd -> upd.API.pool_update_uuid)
+      |> S.of_list in
+    let updates_local = (* patches on local host *)
+      Db.Host.get_updates ~__context ~self:localhost
```

Comment:
We start getting this kind of duplication all over the place, would be nice to be able to pass Db/Client with a common api as function parameters...